src/main/kotlin/com/arsiu/eduhub/model/User.kt

```diff
@@ -1,65 +1,35 @@
 package com.arsiu.eduhub.model
 
 import com.arsiu.eduhub.model.enums.Role
-import com.fasterxml.jackson.annotation.JsonProperty
-import jakarta.persistence.Column
-import jakarta.persistence.Entity
-import jakarta.persistence.EnumType
-import jakarta.persistence.Enumerated
-import jakarta.persistence.GeneratedValue
-import jakarta.persistence.GenerationType
-import jakarta.persistence.Id
-import jakarta.persistence.JoinColumn
-import jakarta.persistence.JoinTable
-import jakarta.persistence.ManyToMany
-import jakarta.persistence.OneToMany
-import jakarta.persistence.Table
+import org.bson.types.ObjectId
+import org.springframework.data.annotation.Id
+import org.springframework.data.mongodb.core.mapping.Document
+import org.springframework.data.mongodb.core.mapping.DocumentReference
 
-@Entity
-@Table(name = "edu_user")
+@Document("user")
 data class User(
 
     @Id
-    @JsonProperty("id")
-    @Column(name = "id")
-    @GeneratedValue(strategy = GenerationType.AUTO)
-    var id: Long = 0L,
+    var id: String = ObjectId().toString(),
```

Comment:
My bad I chose to delegate ID creation to my REST service because:
1. I forgot that I could make it `lateinit`, so it isn't nullable:
```
@Id                                     @Id
var id: String? = null          lateinit var id: String
```
2. I have done this before, so I did it without thinking.
```
@Column(name = "id")
@GeneratedValue(strategy = GenerationType.AUTO)
```
In relational databases, it makes sense because different relational databases can have their own generation strategies. However, in MongoDB, I agree it's better to delegate it to MongoDB.

Reply:
So idk what is the right decision pls help to choose the right one

Reply:
After considering the _difference between storing an ObjectId and its string form_, I choose to use 
**lateinit var id: String** because, after careful consideration of the pros and cons, it appears to be the best option.

Reply:
done

Reply:
After some more investigation if I use lateinit mongo stores references as null
<img width="582" alt="image" src="https://github.com/ArSiu/eduHub/assets/33807305/6e433f03-f4bf-424c-af23-52a39766ed71">

Reply:
done

Reply:
https://lankydan.dev/the-potential-traps-in-kotlins-data-classes

Reply:
done